src/briefcase/console.py

```diff
@@ -250,6 +250,75 @@ def export_log(self):
         """Export the text of the entire log; the log is also cleared."""
         return self._log_impl.export_text()
 
+    def warning_banner(
+        self,
+        message: str,
+        title: str = "",
+        width=80,
+        border_char="*",
+        padding=2,
+    ) -> str:
+        """Format warning banner message inside an asterisk border box.
+
+        :param message: The message to format inside the box.
+        :type message: str
+        :param title: The title of the box. If provided, appears centered at the top.
+        :type title: str, optional
+        :param width: The total width of the box in characters. Defaults to 80.
+        :type width: int, optional
+        :param border_char: Character to use for the box border. Defaults to "*".
+        :type border_char: str, optional
+        :param padding: Number of spaces between the border and text content. Defaults
+            to 2.
+        :type padding: int, optional
+        :return: The formatted message enclosed in a bordered box.
+        :rtype: str
```

Comment:
Apologies for the miscommunication here; but we don't need the type documentation - because they are (or should be) in the type annotations for the method itself.
```suggestion
        :param message: The message to format inside the box.
        :param title: The title of the box. If provided, appears centered at the top.
        :param width: The total width of the box in characters. Defaults to 80.
        :param border_char: Character to use for the box border. Defaults to "*".
        :param padding: Number of spaces between the border and text content. Defaults
            to 2.
        :return: The formatted message enclosed in a bordered box.
```